xonsh/environ.py

```diff
@@ -865,6 +865,14 @@ class GeneralSetting(Xettings):
         "If enabled, the CommandsCache is saved between runs and can reduce the startup time.",
     )
 
+    COMMANDS_CACHE_IGNORE_MTIME_CHECK = Var.with_default(
+        True,
+        "If enabled, ignore mtime check when checking if directories changed"
```

Comment:
- the default can be False. 
- Ultimately this variable disables commands_cache's cache system not just the mtime check. mtime-check is an implementation detail. I think this should be named `ENABLE_COMMANDS_CACHE(default=True)` . I suggest the description to be `command names in a directory are cached when enabled. It makes use of directory's modified time. On some platforms it may not be accurate enough (e.g. Windows, Linux save mtime in seconds). setting it to False would disable the caching mechanism and may slow down the shell`

Reply:
sounds good!
I also changed my test a little bit to run only if the cache is disabled.